### PR TITLE
feat(#319): add registry auth info into Docker manager

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add registry auth info into Docker manager to support pushing and pulling images from registry.

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes partially #319

**Special notes for your reviewer**:

/cc @bbbmj 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/docker/client.go [hunk 1]
```diff
@@ -36,10 +36,12 @@ const (
...
 type DockerManager struct {
 	// Client represets the Docker client.
-	Client *docker_client.Client
+	Client     *docker_client.Client
+	EndPoint   string
+	AuthConfig *docker_client.AuthConfiguration
 }
 
-func NewDockerManager(endpoint string) (*DockerManager, error) {
+func NewDockerManager(endpoint, registryServer, registryUsername, registryPassword string) (*DockerManager, error) {
```
Comment (bbbmj) [inline pkg/docker/client.go]: What's the difference between `endpoint` and `registryServer`?

Comment (supereagle) [reply pkg/docker/client.go]: `endpoint` is Docker server endpoint, which is used to new Docker client.
`registryServer` is the registry server.

**Pull request merged**